tests/pipeline/test_term.py

```diff
@@ -330,6 +334,38 @@ class SomeFactorNoDType(SomeFactor):
         with self.assertRaises(InvalidDType):
             SomeFactor(dtype=1)
 
+    def test_latest_on_different_dtypes(self):
+        factor_dtypes = (int64_dtype, float64_dtype, datetime64ns_dtype)
+        for column in TestingDataSet.columns:
+            if column.dtype == bool_dtype:
+                self.assertIsInstance(column.latest, Filter)
+            elif column.dtype in factor_dtypes:
+                self.assertIsInstance(column.latest, Factor)
+            else:
+                self.fail(
+                    "Unknown dtype %s for column %s" % (column.dtype, column)
+                )
+            # These should be the same value, plus this has the convenient
+            # property of correctly handling `NaN`.
+            self.assertIs(column.missing_value, column.latest.missing_value)
+
+    def test_failure_timing_on_bad_missing_values(self):
```

Comment:
Class construction time seems fine, it is worth it to know which column is incorrect and I imagine that very few (if any) people will be creating columns outside of a dataset.